MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/contacts/activity/ContactsActivityViewModel.kt

```diff
@@ -0,0 +1,757 @@
+        )
+
+        private fun ContactObject.buildDepartmentField() = ContactDetailsField.Department(
+            fieldValue = department,
+            onValueChange = this@DefaultContactDetailsViewModel::onDepartmentChange
+        )
+    }
+
+    private inner class DefaultContactsListViewModel : ContactsListClickHandler {
+
+        private val initialState: ContactsListUiState
+            get() = ContactsListUiState(
+                contacts = emptyList(),
+                curSelectedContactId = null,
+                isDoingInitialLoad = true,
+                isDoingDataAction = false,
+                isSearchJobRunning = false
+            )
+
+        private val mutListUiState = MutableStateFlow(initialState)
+        val uiState: StateFlow<ContactsListUiState> get() = mutListUiState
+
+        val willHandleBack: Boolean = false
+
+        fun reset() {
+            curSearchJob?.cancel()
+            mutListUiState.value = initialState
+        }
+
+        fun onRecordsEmitted() {
+            if (uiState.value.isDoingInitialLoad) {
+                mutListUiState.value = uiState.value.copy(isDoingInitialLoad = false)
+            }
+
+            restartSearch(searchTerm = uiState.value.curSearchTerm) { filteredResults ->
+                mutListUiState.value = uiState.value.copy(contacts = filteredResults)
+            }
+            // TODO handle when selected contact is no longer in the records list
+        }
+
+        fun setSelectedContact(id: String?) {
+            mutListUiState.value = uiState.value.copy(curSelectedContactId = id)
+        }
+
+        override fun contactClick(contactId: String) = safeHandleUiEvent {
+            setContactWithConfirmation(contactId = contactId, editing = false)
+        }
+
+        override fun createClick() = safeHandleUiEvent {
+            setContactWithConfirmation(contactId = null, editing = true)
+        }
+
+        override fun deleteClick(contactId: String) = safeHandleUiEvent {
+            runDeleteOpWithConfirmation(idToDelete = contactId)
+        }
+
+        override fun editClick(contactId: String) = safeHandleUiEvent {
+            setContactWithConfirmation(contactId = contactId, editing = true)
+        }
+
+        override fun undeleteClick(contactId: String) = safeHandleUiEvent {
+            runUndeleteOpWithConfirmation(idToUndelete = contactId)
+        }
+
+        fun onSearchTermUpdated(newSearchTerm: String) = safeHandleUiEvent {
+            mutListUiState.value = uiState.value.copy(curSearchTerm = newSearchTerm)
+
+            restartSearch(searchTerm = newSearchTerm) { filteredList ->
+                eventMutex.withLockDebug {
+                    mutListUiState.value = uiState.value.copy(contacts = filteredList)
+                }
+            }
+        }
+
+        @Volatile
+        private var curSearchJob: Job? = null
+
+        private fun restartSearch(
+            searchTerm: String,
```

Comment:
@gkotula I found a bug.  We need to remove formatting characters ("\n", "\t", etc) from this searchTerm before using it.  

For example, if the user hits enter the search term will always contain the ("\n") newline and will never match any contacts.  The newline is invisible to the user and it is not possible for them to delete it.